services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java

```diff
@@ -97,6 +102,7 @@ public News createNews(News news, org.exoplatform.services.security.Identity cur
       News createdNews;
       if (PublicationDefaultStates.PUBLISHED.equals(news.getPublicationState())) {
         createdNews = postNews(news, currentIdentity.getUserId());
+        newsTargetingService.linkNewsToTargets(news);
```

Comment:
I think that you should call instead removeNewsTargets then setNewsTargets in publishNews method